src/Umbraco.Core/Services/NotificationService.cs

```diff
@@ -204,6 +204,8 @@ public Notification CreateNotification(IUser user, IEntity entity, string action
             if (createSubject == null) throw new ArgumentNullException("createSubject");
             if (createBody == null) throw new ArgumentNullException("createBody");
 
+            allVersions = allVersions.OrderBy(x => x.UpdateDate).ToArray();
```

Comment:
Yeah I actually did that at first too, but then realized I wasn't really sure if this would have any implications on other parts of the system. 
Currently the versions come out with newest first as it is, so changing that default behavior might have implications elsewhere. Seeing as this service was pretty much relying on the order to just be as expected, I'd suspect other parts of the system might do so too.